### DISCUSSION TITLE
Welcome to sherpa-rs Discussions!

### DISCUSSION THREAD

**Discussion by thewh1teagle** [thewh1teagle/sherpa-rs#51] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by ghjdegithub** [comment]
I'm going to go up the lifecycle, and use the chat* pointer returned by result directly, without copying the string

**Reply by thewh1teagle** [reply]
Start with very specific and tiny PR to specific struct so we can review

**Reply by ghjdegithub** [reply]
> Start with very specific and tiny PR to specific struct so we can review 

You merge my previous one, and I'm going to add a lifecycle on top of that, and a chat* that returns directly using the C API, which will also add a lifecycle
